src/components/Checkbox.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+
+import { db } from '../lib/firebase';
+
+const Checkbox = ({ item, shoppingList }) => {
+  const [checked, setChecked] = useState(false);
+
+  const { name, id, lastPurchasedDate } = item;
+
+  const isExpired = (date) => {
+    const expiryDate = item.lastPurchasedDate + 60 * 60 * 24;
+    const now = new Date().getTime();
+    return expiryDate < now;
+  };
+
+  useEffect(() => {
+    if (lastPurchasedDate === null) {
+      setChecked(false);
+    }
+    if (isExpired(lastPurchasedDate)) {
+      setChecked(false);
+    } else {
+      setChecked(true);
+    }
```

Comment:
very clear and readable here -- love it

Reply:
Hi @alexlsalt, thank you. How could the isExpired function be updated?

Reply:
I think the comment is on the checkbox file:

![9ACA7F3B-0F88-4FF3-8725-098289505492](https://user-images.githubusercontent.com/45923677/116753651-b39f8100-aa07-11eb-8a66-b52a2319106d.png)

Reply:
hi @alexlsalt, my logic in the isExpired function was correct. What I forgot was that we're dealing with milliseconds and that my value in the expiry date variable needed to be multiplied by another 1000. It's all fixed now and is WORKING!!! :muscle:

Reply:
yaaaasssss! 🎉 🎉 🎉 🎉 🎉